rclcpp/include/rclcpp/client.hpp

```diff
@@ -195,6 +201,88 @@ class Client : public ClientBase
 
   RCLCPP_SMART_PTR_DEFINITIONS(Client)
 
+  /// A convenient Client::Future and request id pair.
+  class FutureAndRequestId
+  {
+public:
+    FutureAndRequestId(Future impl, int64_t req_id)
+    : impl_(std::move(impl)), req_id_(req_id)
+    {}
+
+    /// Allow implicit conversions to `std::future` by reference.
+    // TODO(ivanpauno): Maybe, deprecate this in favor of get_future() (?)
+    operator Future &() {return impl_;}
+
+    /// Deprecated, use take_future() instead.
+    /**
+     * Allow implicit conversions to `std::future` by value.
+     * \deprecated
+     */
+    [[deprecated("FutureAndRequestId: use take_future() instead of an implicit conversion")]]
+    operator Future() {return impl_;}
```

Comment:
I think I actually don't need this two custom implicit conversions, but actually one to `SharedFuture` to avoid breaking pre-existing code.
I'm taking a look locally.